File: VB/WebSite/App_Code/MyDataSource.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web.UI
Public Class MyDataSource
    Implements IHierarchicalDataSource

    Public Event DataSourceChanged As EventHandler ' This event is never used, but it's necessary for correct implementation of the IHierarchialDataSource interface

    Public Function GetHierarchicalView(ByVal viewPath As String) As HierarchicalDataSourceView
        Return New MyHierarchicalView()
    End Function
End Class

File: CS/WebSite/App_Code/MyDataSource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.UI;
public class MyDataSource : IHierarchicalDataSource
{
    public event EventHandler DataSourceChanged; // This event is never used, but it's necessary for correct implementation of the IHierarchialDataSource interface

    public HierarchicalDataSourceView GetHierarchicalView(string viewPath)
    {
        return new MyHierarchicalView();
    }
}